### PR TITLE
Support KSDeclaration.getDocString

### PR DESCRIPTION
**Pull request by ting-yuan**
A doc string is a string in the comment right in front of an declaration
and is enclosed by /** and */. Leading whitespaces and * are removed for
each line.

### CODE REVIEW

File: compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/utils.kt [hunk 1]
```diff
@@ -226,6 +227,21 @@ fun PsiElement.toLocation(): Location {
...
+    if (!t1.startsWith("/**") || !t1.endsWith("*/"))
+        return null
+    val lineSep = t1.findAnyOf(listOf("\r\n", "\n", "\r"))?.second ?: ""
+    return t1.trim('/').trim('*').lines().joinToString(lineSep) {
+        it.trimStart().trimStart('*')
+    }
+}
+
+fun PsiElement.getDocString(): String? =
+    this.firstChild.siblings().firstOrNull { it is PsiComment }?.let {
```
Comment (neetopia) [inline compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/utils.kt:241]: Should we take care of the case when normal comments are used with document comments?

Comment (ting-yuan) [reply compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/utils.kt:241]: What do you mean by normal comments? `//`, `/* */` or ...?

Comment (neetopia) [reply compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/utils.kt:241]: both

Comment (ting-yuan) [reply compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/utils.kt:241]: No, we should keep it simple and not include random comments.

Comment (neetopia) [reply compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/utils.kt:241]: Consider this code:
```
// comment 1
/**
* document 
*/ 
class Foo
```
what will happen with your implementation?

Comment (ting-yuan) [reply compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/utils.kt:241]: comment 1 will be ignored.

**Pull request merged**